cmd/state-installer/cmd.go

```diff
@@ -272,10 +265,14 @@ func execute(out output.Outputer, cfg *config.Instance, an analytics.Dispatcher,
 	// Detect whether this is a fresh install or an update
 	isUpdate := false
 	switch {
+	case (params.sourceInstaller == "install.sh" || params.sourceInstaller == "install.ps1") && fileutils.FileExists(packagedStateExe):
+		logging.Debug("Not using update flow as installing via " + params.sourceInstaller)
+		params.sourcePath = installerPath
```

Comment:
I like this change, as it makes the behavior more explicit and less error prone.

That said, this has surfaced another bug in our code. When I run:

```
./installers/install.sh -b dx-806
```

I get:

```
Installation path must be an empty directory
```

It seems we might be targeting the wrong directory because I've never installed this branch before. Previously this probably didn't trigger any errors because it considered it an "update".

I just checked, and it does indeed seem we might be targeting the wrong install dir:

```
❯ which state
/Users/nathanrijksen/.local/ActiveState/StateTool/bin/state
```

That should have been:

```
/Users/nathanrijksen/.local/ActiveState/StateTool/beta/bin/state
```

Note the first variant is missing the branch.

Reply:
I realize this is becoming a bit of a scope creep for your story. If you prefer we can file a separate ticket for this issue.

Reply:
Strange, I'm not seeing that. `-b dx-806` works just fine for me:

```
mitchellb@ManBearPig cli % ls ~/.local/ActiveState/StateTool
release
mitchellb@ManBearPig cli % which state
/Users/mitchellb/.local/ActiveState/StateTool/release/bin/state
mitchellb@ManBearPig cli % state --version
ActiveState CLI by ActiveState Software Inc.
License BSD 3
Version 0.33.0-SHA7b42690
Revision 7b4269018813827216b40f975aca606e02e1b4de
Branch release
Built Tue Mar 22 2022 20:18:03 +0000 UTC

mitchellb@ManBearPig cli % ./installers/install.sh -b dx-806
• Preparing Installer for State Tool Package Manager version 0.34.0-SHA6f6f995... ✔ Done

█ Installing State Tool Package Manager

The State Tool lets you install and manage your language runtimes.

ActiveState collects usage statistics and diagnostic data about failures. 
By using the State Tool Package Manager you agree to the terms of ActiveState’s Privacy Policy, 
available at: https://www.activestate.com/company/privacy-policy

• Installing State Tool to /Users/mitchellb/.local/ActiveState/StateTool/dx-806... ✔ Done

█ State Tool Package Manager Installation Complete

State Tool Package Manager has been successfully installed.
mitchellb@ManBearPig cli % which state
/Users/mitchellb/.local/ActiveState/StateTool/dx-806/bin/state
mitchellb@ManBearPig cli % state --version
ActiveState CLI by ActiveState Software Inc.
License BSD 3
Version 0.34.0-SHA6f6f995
Revision 6f6f99565390592ab07fb8abb9ea11a470b90521
Branch dx-806
Built Wed Apr 13 2022 21:43:28 +0000 UTC
```

Reply:
I think I had a corrupted install somehow. I reinstalled fresh and then tried again, this time I did not get a repro.